sdks/python/apache_beam/runners/portability/fn_api_runner.py

```diff
@@ -813,20 +892,29 @@ def state_api_service_descriptor(self):
     return endpoints_pb2.ApiServiceDescriptor(
         url='localhost:%s' % self.state_port)
 
+  def logging_api_service_descriptor(self):
+    return endpoints_pb2.ApiServiceDescriptor(
+        url='localhost:%s' % self.logging_port)
+
   def close(self):
     self.control_handler.done()
     self.data_plane_handler.close()
-    self.control_server.stop(5).wait()
-    self.data_server.stop(5).wait()
-    self.state_server.stop(5).wait()
+    to_wait = [
+        self.control_server.stop(5),
+        self.data_server.stop(5),
+        self.state_server.stop(5),
+        self.logging_server.stop(5)
```

Comment:
I assume this is a timeout duration--can we factor that out into a constant?

Reply:
Done.